### PR TITLE
Document/clarify code, add (explicit) constructor from Coords to TileCoords

### PR DESCRIPTION
**Pull request by NickNick**
While not part of something bigger, while reading through the code and trying to clear things up I made these changes I thought were worthwhile to keep.

### CODE REVIEW

File: src/openrct2/world/Map.cpp [hunk 1]
```diff
@@ -729,10 +729,10 @@ bool map_coord_is_connected(sint32 x, sint32 y, sint32 z, uint8 faceDirection)
             continue;
 
         rct_tile_element_path_properties props = tileElement->properties.path;
-        uint8 pathType = props.type >> 2;
+        bool sloped = (props.type >> 2) & 1;
```
Comment (Gymnasiast) [inline src/openrct2/world/Map.cpp]: `footpath_is_sloped(tileElement);`

File: src/openrct2/peep/Peep.cpp [hunk 2]
```diff
@@ -10754,7 +10754,7 @@ sint32 peep_pathfind_choose_direction(sint16 x, sint16 y, uint8 z, rct_peep * pe
              * In the future these could be used to visualise the
              * pathfinding on the map. */
             uint8          endJunctions         = 0;
-            TileCoordsXYZ endJunctionList[16]  = { 0 };
+            TileCoordsXYZ endJunctionList[16];
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: This is not needed.

File: src/openrct2/world/Location.hpp [hunk 3]
```diff
@@ -52,24 +52,30 @@ constexpr sint32 COORDS_NULL = -1;
  * Tile coordinates use 1 x/y increment per tile.
  * Regular ('big', 'sprite') coordinates use 32 x/y increments per tile.
  * */
-struct TileCoordsXY
+struct CoordsXY
```
Comment (Gymnasiast) [inline src/openrct2/world/Location.hpp]: Why did you reverse the two? It only adds noise to the diff.

File: src/openrct2/interface/Colour.h [hunk 4]
```diff
@@ -154,4 +154,6 @@ void colours_init_maps();
 uint8 blendColours(const uint8 paletteIndex1, const uint8 paletteIndex2);
 #endif
 
+uint8 findClosestPaletteIndex(uint8 red, uint8 green, uint8 blue);
```
Comment (Gymnasiast) [inline src/openrct2/interface/Colour.h]: The point of a commit message is to tell why you did this.

**Comment by Gymnasiast** [comment]
This seems to be a collection of unrelated changes. The commit messages are not really descriptive either, and miss explanation as to why you made the change.

I hope I'm not coming across as too harsh (especially seeing as this is your first commit).

File: src/openrct2/world/Location.hpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/openrct2/world/Location.hpp]: Nick's new definitions in TileCoordsXY below require CoordsXY to be defined at an earlier point.

File: src/openrct2/interface/Colour.h (hunk 4 above)
Comment (AaronVanGeffen) [reply src/openrct2/interface/Colour.h]: I believe this was done to expose the function for Nick and Jaxan's upcoming minimap feature. See Gitter.

File: src/openrct2/world/Location.hpp (hunk 3 above)
Comment (Broxzier) [reply src/openrct2/world/Location.hpp]: The diff just looks confusing. `CoordsXY` has been moved to be above `TileCoordsXY`

File: src/openrct2/interface/Colour.h (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/interface/Colour.h]: My point is that the commit message/description should make that clear.

File: src/openrct2/world/Location.hpp [hunk 5]
```diff
@@ -52,24 +52,30 @@ constexpr sint32 COORDS_NULL = -1;
...
     sint32 x, y, z;
 };
 
-struct CoordsXYZ
+struct TileCoordsXYZ
 {
-    sint32 x, y, z;
+    TileCoordsXYZ() = default;
+    TileCoordsXYZ(sint32 x_, sint32 y_, sint32 z_) : x(x_), y(y_), z(z_) {}
+    explicit TileCoordsXYZ(CoordsXYZ c) : x(c.x/32), y(c.y/32), z(c.z/32) {}
```
Comment (Gymnasiast) [inline src/openrct2/world/Location.hpp]: It's `z / 8`, not `z / 32`.

File: src/openrct2/interface/Colour.h (hunk 4 above)
Comment (AaronVanGeffen) [reply src/openrct2/interface/Colour.h]: I don't disagree. You could even make the argument that the change can be introduced at a later point, e.g. in the eventual PR that makes it relevant.

File: src/openrct2/peep/Peep.cpp (hunk 2 above)
Comment (NickNick) [reply src/openrct2/peep/Peep.cpp]: It is with the other change. There's no ctor that accepts '0' as parameter, so...

File: src/openrct2/world/Location.hpp (hunk 3 above)
Comment (NickNick) [reply src/openrct2/world/Location.hpp]: What @AaronVanGeffen said.

File: src/openrct2/interface/Colour.h (hunk 4 above)
Comment (NickNick) [reply src/openrct2/interface/Colour.h]: It's to make upcoming PRs smaller, yeah. I didn't see any harm in exposing it. It's a useful function? I dunno.. seemed like a small thing. I can delay it.

File: src/openrct2/world/Location.hpp (hunk 5 above)
Comment (NickNick) [reply src/openrct2/world/Location.hpp]: Thanks! 👍

**Comment by NickNick** [comment]
Nothing's too harsh for my software engineer soul! Fair points. I guess and alter my current develop and send another pull request?

**Comment by AaronVanGeffen** [comment]
Cheers. No need to create a new PR. If you amend and push your branch, this PR will be updated automatically. :)

**Comment by NickNick** [comment]
There's no real 'why' to 1dbfd1d and eb57fd38: they just try to make the code more readable but that is sort of implied by the 'Document...' ? Open to suggestions if @Gymnasiast wants to see that fixed/described better.

**Comment by AaronVanGeffen** [comment]
We do like the commit history to be reasonably clean, so this looks fine to me. You might want to squash the documentation commits together as well.

File: src/openrct2/world/Location.hpp [hunk 6]
```diff
@@ -52,24 +52,30 @@ constexpr sint32 COORDS_NULL = -1;
...
     sint32 x, y;
 };
 
-struct CoordsXY
+struct TileCoordsXY
 {
-    sint32 x, y;
+    TileCoordsXY() = default;
+    TileCoordsXY(sint32 x_, sint32 y_) : x(x_), y(y_) {}
+    explicit TileCoordsXY(CoordsXY c) : x(c.x/32), y(c.y/32) {}
```
Comment (Gymnasiast) [inline src/openrct2/world/Location.hpp]: Needs its formatting adjusted:
`explicit TileCoordsXY(CoordsXY c) : x(c.x / 32), y(c.y / 32) {}`

File: src/openrct2/world/Location.hpp [hunk 7]
```diff
@@ -52,24 +52,30 @@ constexpr sint32 COORDS_NULL = -1;
...
     sint32 x, y, z;
 };
 
-struct CoordsXYZ
+struct TileCoordsXYZ
 {
-    sint32 x, y, z;
+    TileCoordsXYZ() = default;
+    TileCoordsXYZ(sint32 x_, sint32 y_, sint32 z_) : x(x_), y(y_), z(z_) {}
+    explicit TileCoordsXYZ(CoordsXYZ c) : x(c.x/32), y(c.y/32), z(c.z/8) {}
```
Comment (Gymnasiast) [inline src/openrct2/world/Location.hpp]: See previous point.

**Pull request merged**